testing.rst

```diff
@@ -1137,12 +1137,30 @@ Mailer Assertions
     address. This assertion normalizes addresses like ``Jane Smith
     <jane@example.com>`` into ``jane@example.com``.
 
-.. TODO
-..  End to End Tests (E2E)
-..  ----------------------
-..  * panther
-..  * testing javascript
-..  * UX or form collections as example?
+End to End Tests (E2E)
+~~~~~~~~~~~~~~~~~~~~~~
+
+End to end tests are a special type of application tests that
+simulate a real user interacting with your application. They are
+typically used to test the user experience (UX) of your application.
+The difference with functional tests detailed above is that End-to-End tests
+use a real browser instead of a simulated one. This browser can run in
+headless mode (without a graphical interface) or not. The first option
+is convenient for running tests in a Continuous Integration (CI), while
+the second one is useful for debugging purpose.
+
+This is the purpose of `Panther`_, a library that provides a real browser
+to run your tests. Here are a few things that make Panther special, compared
+to other End to End testing tools:
```

Comment:
```suggestion
to other testing tools provided by Symfony:
```

Because most, if not all, E2E testing tools have the capabilities described below.